convert2rhel/pkghandler.py

```diff
@@ -273,43 +299,69 @@ def get_installed_pkgs_by_fingerprint(fingerprints, name=""):
     # possible, converted. This issue does not happen on yum, so we can still
     # use only the package name for it.
     return [
-        "%s.%s" % (pkg.pkg_obj.name, pkg.pkg_obj.arch) for pkg in pkgs_w_fingerprints if pkg.fingerprint in fingerprints
+        "%s.%s" % (pkg.nevra.name, pkg.nevra.arch) for pkg in pkgs_w_fingerprints if pkg.fingerprint in fingerprints
     ]
 
 
-def get_installed_pkgs_w_fingerprints(name=""):
-    """Return a list of objects that hold one of the installed packages (yum.rpmsack.RPMInstalledPackage in case
-    of yum and hawkey.Package in case of dnf) and GPG key fingerprints used to sign it. The packages can be
-    optionally filtered by name.
-    """
-    package_objects = get_installed_pkg_objects(name)
-    pkgs_w_fingerprints = []
-    for pkg_obj in package_objects:
-        fingerprint = get_pkg_fingerprint(pkg_obj)
-        pkgs_w_fingerprints.append(PkgWFingerprint(pkg_obj, fingerprint))
+def _get_pkg_fingerprint(signature):
+    """Get fingerprint of the key used to sign a package."""
+    fingerprint_match = re.search("Key ID (.*)", signature)
+    return fingerprint_match.group(1) if fingerprint_match else "none"
 
-    return pkgs_w_fingerprints
 
+def get_installed_pkg_information(pkg_name=""):
+    """
+    Get information about a package, such as signature from the RPM database,
+    packager, vendor, NEVRA and fingerprint.
 
-def get_pkg_fingerprint(pkg_obj):
-    """Get fingerprint of the key used to sign a package."""
-    pkg_sig = get_pkg_signature(pkg_obj)
-    fingerprint_match = re.search("Key ID (.*)", pkg_sig)
-    if fingerprint_match:
-        return fingerprint_match.group(1)
+    :param pkg_name: Full name of a package to check their signature.  If not given, information about all installed packages is returned.
+    :type pkg_obj: str
+    :return: Return the package signature.
+    :rtype: list[PackageInformation]
+    """
+    cmd = [
+        "rpm",
+        "--qf",
+        "C2R %{PACKAGER}&%{VENDOR}&%{NAME}-%|EPOCH?{%{EPOCH}}:{0}|:%{VERSION}-%{RELEASE}.%{ARCH}&%|DSAHEADER?{%{DSAHEADER:pgpsig}}:{%|RSAHEADER?{%{RSAHEADER:pgpsig}}:{%|SIGGPG?{%{SIGGPG:pgpsig}}:{%|SIGPGP?{%{SIGPGP:pgpsig}}:{(none)}|}|}|}|\n",
+    ]
+
+    if pkg_name:
+        if "*" in pkg_name:
+            cmd.append("-qa")
+        else:
+            cmd.extend(["-q", pkg_name])
     else:
```

Comment:
```suggestion
    if pkg_name:
        if "*" in pkg_name:
            cmd.append(["-qa", pkg_name])
        else:
            cmd.extend(["-q", pkg_name])
    else:
```

or, if the default of pkg_name is changed to `"*"`, then:
``` diff
-    if pkg_name:
-        if "*" in pkg_name:
-            cmd.append("-qa")
-        else:
-            cmd.extend(["-q", pkg_name])
+    if "*" in pkg_name:
+        cmd.extend(["-qa", pkg_name])
+    else:
+        cmd.extend(["-q", pkg_name])
-    else:
-        cmd.append("-qa")
```

Reply:
I think that like this would be more correct, no? 

```python
    if pkg_name:
        if "*" in pkg_name:
            cmd.extend(["-qa", pkg_name])
        else:
            cmd.extend(["-q", pkg_name])
    else:
        cmd.append("-qa")
```

Reply:
updated